src/views/virtualmachinesinstance/details/tabs/disks/hooks/useFileSystemTableGuestOS.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+import VirtualMachineInstanceModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineInstanceModel';
+import {
+  V1VirtualMachineInstance,
+  V1VirtualMachineInstanceGuestAgentInfo,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { consoleFetch } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseFileSystemTableGuestOS = (
+  vmi: V1VirtualMachineInstance,
+) => [V1VirtualMachineInstanceGuestAgentInfo, boolean, any];
+
+const useFileSystemTableGuestOS: UseFileSystemTableGuestOS = (vmi) => {
+  const [loaded, setLoaded] = React.useState(false);
+  const [data, setData] = React.useState<V1VirtualMachineInstanceGuestAgentInfo>({});
+  const [error, setError] = React.useState(null);
+
+  React.useEffect(() => {
+    const guestOS = vmi?.status?.guestOSInfo?.id;
+
+    const fetchingData = async () => {
```

Comment:
can you declare this outside of the `useEffect`

Reply:
if it will be outside we need to add as a dependency, what benefit we will get if it's outside?